part/set.go

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package part
+
+type Set[T any] struct {
```

Comment:
Added comment. 

Hmm not sure how useful stating the (vorst case) algorithmic complexity here is. It can in worst case be `O(k)` as radix trees devolve into essentially linked lists for degenerate cases like storing "aaaa" ... "zzzz": you'd have nodes a->a->a->a, but of course in practice it'd be sparse and more like `O(k log n)`.

Reply:
Yeah, that's a fair point. Good enough as is!